pkg/importer/vddk-datasource.go

```diff
@@ -248,18 +248,24 @@ func createVMwareClient(endpoint string, accessKey string, secKey string, thumbp
 	}
 
 	// Construct VMware SDK URL and get MOref
-	sdkURL := vmwURL.Scheme + "://" + accessKey + ":" + secKey + "@" + vmwURL.Host + "/sdk"
+	sdkURL := vmwURL.Scheme + "://" + url.PathEscape(accessKey) + ":" + url.PathEscape(secKey) + "@" + vmwURL.Host + "/sdk"
+	redacted := vmwURL.Scheme + "://" + url.PathEscape(accessKey) + ":" + "*****" + "@" + vmwURL.Host + "/sdk"
 	vmwURL, err = url.Parse(sdkURL)
 	if err != nil {
-		klog.Errorf("Unable to create VMware URL: %v", err)
-		return nil, err
+		return nil, errors.New("unable to create VMware URL from: " + redacted)
 	}
 
 	// Log in to vCenter
 	ctx, cancel := context.WithCancel(context.Background())
 	conn, err := govmomi.NewClient(ctx, vmwURL, true)
 	if err != nil {
-		klog.Errorf("Unable to connect to vCenter: %v", err)
+		if secKey != "" {
+			wrap := url.PathEscape(accessKey) + ":%s@"
+			target := fmt.Sprintf(wrap, url.PathEscape(secKey))
+			redacted = fmt.Sprintf(wrap, "*****")
+			redacted = strings.ReplaceAll(err.Error(), target, redacted)
+			err = errors.New("unable to connect to vCenter: " + redacted)
+		}
```

Comment:
using `vmwURL.Redacted()` should be safer.

Reply:
With the previous fix, this whole error branch is no longer necessary. The password isn't part of the bad URL string anymore, so the error from govmomi is now just "incorrect username or password". Also much better.